src/bluesearch/entrypoint/database/topic_extract.py

```diff
@@ -151,57 +146,38 @@ def run(
         print(*inputs, sep="\n")
         return 0
 
+    try:
+        article_source = ArticleSource(source)
+    except ValueError:
+        logger.error("Unknown article source: %s", source)
+        return 1
```

Comment:
I think it would be cleaner to handle at arg-parsing time this casting of the `str` passed on the command line into an `ArticleSource`.

This means the following small changes:
1. Specify the `type` in the `ArgumentParser`:
    ```python
    parser.add_argument(
        "source",
        type=ArticleSource,
        choices=(
            "arxiv",
            "biorxiv",
            "medrxiv",
            "pmc",
            "pubmed",
        ),
        choices=[member.value for member in ArticleSource.__members__.values()],
    ```
3. Change the signature of `run()` to take an `ArticleSource`:
    ```python
   def run(
      *,
      source: ArticleSource,
      input_path: Path,
      output_file: Path,
      match_filename: str | None,
      recursive: bool,
      overwrite: bool,
      dry_run: bool,
     ) -> int:
   ```
3. Remove the then unnecessary `try... except` block here.

Reply:
Yes, makes total sense. However, it seems that for some reason `argparse` does not play well with enums :( The use of enums is explicitly discouraged [in the docs](https://docs.python.org/3/library/argparse.html#choices). There's also a discussion about it [in the bug tracker](https://bugs.python.org/issue25061).

Long story short, this:
```python
parser.add_argument(
    "source",
    type=ArticleSource,
    choices=[member.name for member in ArticleSource],
```
doesn't work because argparse first applies the callable in the `type` argument and only then checks against the `choices`, which never matches because it compares `str` to `ArticleSource`.

This:
```python
parser.add_argument(
    "source",
    type=ArticleSource,
    choices=ArticleSource,
```
works, but the help message is misleading:
```
 bbs_database topic-extract --help
usage: bbs_database topic-extract [-h] [-v] [-m MATCH_FILENAME] [-R] [-o] [-n]
                                  {ArticleSource.ARXIV,ArticleSource.BIORXIV,ArticleSource.MEDRXIV,ArticleSource.PMC,ArticleSource.PUBMED,ArticleSource.UNKNOWN}
                                  input_path output_file
```
So one might have to stick with `str` type and casting to `ArticleSource` later. The `try-except` is probably not necessary if we use `choices`.

Reply:
I removed the `try-except` in b1df1ae (+some cleaning up), but didn't didn't add `type=ArticleSource` otherwise.

Reply:
Thank you for the explanation, I had tried it on my laptop and I did not notice that you indeed get misleading help messages for `choices=ArticleSource`.

I tried setting `ArticleSource.__repr__` but it also does not work! 🤷